Frameworks/CoreText/CTRun.mm

```diff
@@ -24,52 +24,36 @@
 #import "CoreTextInternal.h"
 #import "CGContextInternal.h"
 #import <CppUtils.h>
-
-#include <numeric>
-#include <algorithm>
-#include <functional>
+#import <numeric>
+#import <algorithm>
+#import <functional>
 
 const CFStringRef kCTBackgroundStrokeColorAttributeName = CFSTR("kCTBackgroundStrokeColorAttributeName");
 const CFStringRef kCTBackgroundFillColorAttributeName = CFSTR("kCTBackgroundFillColorAttributeName");
 const CFStringRef kCTBackgroundCornerRadiusAttributeName = CFSTR("kCTBackgroundCornerRadiusAttributeName");
 const CFStringRef kCTBackgroundLineWidthAttributeName = CFSTR("kCTBackgroundLineWidthAttributeName");
 
-@implementation _CTRun : NSObject
-- (void)dealloc {
-    _stringFragment = nil;
-    _attributes = nil;
-    _SafeRelease(&_dwriteGlyphRun.fontFace);
-    delete[] _dwriteGlyphRun.glyphIndices;
-    delete[] _dwriteGlyphRun.glyphAdvances;
-    delete[] _dwriteGlyphRun.glyphOffsets;
-    [super dealloc];
-}
-
-- (id)init {
-    if (self = [super init]) {
-        _attributes.attach([NSMutableDictionary new]);
-        _textMatrix = CGAffineTransformIdentity;
-    }
-
-    return self;
+// We need to include DWriteWrapper_CoreText.h for _CloneDWriteGlyphRun,
+// which is located in the Frameworks\CoreText folder
+// Given how the earlier implementation is accessing and assigning internal variables in different parts of this package
+// struct __CTRun has to be made public via CoreTextInternal.h (Frameworks\include). Therefore __CTRun(const CTRunRef& other)'s
+// implementation needs to be moved here.
+__CTRun::__CTRun(const CTRunRef& other)
+    : _attributes(CFDictionaryCreateMutableCopy(kCFAllocatorSystemDefault, CFDictionaryGetCount(other->_attributes), other->_attributes)),
```

Comment:
Yep, I forgot that the bug is now fixed :p

---
In reply to: [111453534](https://github.com/Microsoft/WinObjC/pull/2511#discussion_r111453534) [](ancestors = 111453534)